### PR TITLE
Feature: more convenient uploads

### PR DESCRIPTION
**Pull request by lsh-0**
* adds node parameter to 'upload_file' task
* adds a default destination if one not specified

### CODE REVIEW

File: src/cfn.py [hunk 1]
```diff
@@ -300,9 +301,10 @@ def _download(path, destination):
 
 @task
 @requires_aws_stack
-def upload_file(stackname, local_path, remote_path, overwrite=False):
-    with stack_conn(stackname):
-        print('stack:', stackname)
+def upload_file(stackname, local_path, remote_path=None, overwrite=False, node=1):
+    remote_path = remote_path or os.path.join("/tmp", os.path.basename(local_path))
+    with stack_conn(stackname, node=node):
```
Comment (giorgiosironi) [inline src/cfn.py]: didn't remember this supported `node`

It seems it is forced to `int` inside the function (`node` here will be a string)

File: src/cfn.py [hunk 2]
```diff
@@ -300,9 +301,10 @@ def _download(path, destination):
...
+        print('stack:', stackname, 'node', node)
```
Comment (giorgiosironi) [inline src/cfn.py]: maybe we should start using a logger here

File: src/cfn.py (hunk 1 above)
Comment (lsh-0) [reply src/cfn.py]: > didn't remember this supported node

`stack_conn` supporting a `node` parameter is something I introduced relatively recently - this year some time. I think it was so I could ssh into a specific node in one command

good catch with the typing. I'll fix that up

File: src/cfn.py (hunk 2 above)
Comment (lsh-0) [reply src/cfn.py]: it's not a debug statement, it's just for the user. if anything it should be on stderr. if we have a logger that doesn't emit any prefix that might be ok, but the benefits of that would be quite tiny.

**Pull request merged**